### DISCUSSION TITLE
Multiple Apps in single project using _app.tsx

### DISCUSSION THREAD

**Discussion by MohammedAliBinTaleb** [vercel/next.js#41659] [Ideas]
### **I think we need in next release features that allow us to create multiple apps in single project 😲**

**The Problem :** when I want to create state management using react context 
I have to make every provider in _app.tsx root directory of pages folder
 and that maybe not good because Imagine that state i just live in for example
**/admin/dashboard** route and every routes that come inside this route should access my state and my logic only 
not the whole app like **community pages**  or  **login pages should have access dashboard state **
 
**my simple solution would look like : ** 
just adding /admin/dashboard/_app.tsx so we can add our provider inside it🥳

I hope I can be members in next.js community 😅

**Comment by thiswallz** [comment]
if you dont need to share among components , that means you don't need to have an state such as you mentioned, state provider  is for sharing,.
 with that approach you will be breaking "single source of true" too, and would be a different pattern
